packages/amplify_api_rest/android/src/main/kotlin/com/amazonaws/amplify/amplify_api_rest/AmplifyApiRestPlugin.kt

```diff
@@ -0,0 +1,311 @@
+ */
+
+package com.amazonaws.amplify.amplify_api_rest
+
+import android.os.Handler
+import android.os.Looper
+import androidx.annotation.NonNull
+import androidx.annotation.VisibleForTesting
+import com.amazonaws.amplify.amplify_api_rest.types.FlutterAuthFailureMessage
+import com.amazonaws.amplify.amplify_api_rest.types.FlutterRestInputs
+import com.amazonaws.amplify.amplify_api_rest.types.FlutterRestResponse
+import com.amplifyframework.AmplifyException
+import com.amplifyframework.api.ApiException
+import com.amplifyframework.api.aws.AWSApiPlugin
+import com.amplifyframework.api.rest.RestOperation
+import com.amplifyframework.api.rest.RestOptions
+import com.amplifyframework.api.rest.RestResponse
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+import kotlin.reflect.KFunction3
+import kotlin.reflect.KFunction4
+
+class AmplifyApiRestPlugin: FlutterPlugin, MethodCallHandler {
+
+  private lateinit var channel : MethodChannel
+  private var operationsMap : HashMap<String, RestOperation> = HashMap()
+
+  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "com.amazonaws.amplify/amplify_api_rest")
+    channel.setMethodCallHandler(this)
+    Amplify.addPlugin(AWSApiPlugin())
+  }
+
+  companion object {
+    val LOG = Amplify.Logging.forNamespace("amplify:flutter:api_rest")
+
+    @JvmStatic
+    fun registerWith(registrar: Registrar) {
+      val channel = MethodChannel(registrar.messenger(), "com.amazonaws.amplify/amplify_api_rest")
+      channel.setMethodCallHandler(AmplifyApiRestPlugin())
+      Amplify.addPlugin(AWSApiPlugin())
+      LOG.info("Added AWSApiPlugin")
+    }
+
+  }
+
+  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+    channel.setMethodCallHandler(null)
+  }
+
+  @VisibleForTesting
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+    
+    if(call.method == "cancel"){
+      onCancel(result, (call.arguments as String))
+      return
+    }
+    
+    var arguments : Map<String, Any>
+
+    try {
+      arguments = checkArguments(call.arguments as Map<String, Any>)
+    }
+    catch(e: java.lang.Exception){
+      result.error("WrongMapInput", "The provided map input was not correct", e.message)
+      return
+    }
+    
+    when (call.method) {
+      "get" -> onGet(result, arguments)
+      "post" -> onPost(result, arguments)
+      "put" -> onPut(result, arguments)
+      "delete" -> onDelete(result, arguments)
```

Comment:
I suspect that iOS and Android currently support a different set of HTTP verbs from one-another. But Android, at least, supports a few more than what you have here right now.

Reply:
Thanks! It looks like both Android and iOS have the "head" and "patch" methods so I need to add them in.  It's strange that there's no documentation on the Amplify website for this?